segmentation/utils/segmentation_utils.py

```diff
@@ -278,6 +278,86 @@ def combine_segmentation_masks(big_mask_xr, small_mask_xr, size_threshold,
         format="NETCDF3_64BIT")
 
 
+def find_nuclear_mask_id(nuc_segmentation_mask, cell_coords):
+    """Get the ID of the nuclear mask which has the greatest amount of overlap with a given cell
+
+    Args:
+        nuc_segmentation_mask: label mask of nuclear segmentations
+        cell_coords: list of coords specifying pixels that belong to a cell
+
+    Returns:
+        nuclear_mask_id: ID of the nuclear mask that overlaps most with cell. If not matches found,
+            returns None.
+    """
+
+    ids, counts = np.unique(nuc_segmentation_mask[tuple(cell_coords.T)], return_counts=True)
+
+    max_id = ids[np.argmax(counts)]
```

Comment:
Not sure if it matters, but if there are equal counts of many different id's, `np.argmax(counts)` will only return first-in-order id.

Reply:
For now we're not going to deal with cells that have multiple nuclei, or are poorly segmented and could potentially have multiple. But something to think about for the future.